### PR TITLE
Fix: GitHub issue link is to be displayed on the task details page.

### PR DESCRIPTION
**Pull request by kapillondhe**
# What is the change?
Changes in the task validator allow the GitHub issue link from the client request. This will store the GitHub issue html URL in the task database. This link will be further used in website-status to open the git issue from the Task details page.

# Is Development Tested?

-  [x] Yes
-  [ ] No

# Note to Reviewers:
PR 827 raised in website-status for client related changes.

# Test stat
Before : 
image

After : 
image

### CODE REVIEW

**Comment by sahsisunny** [comment]
Please add the screenshot of test coverage

**Comment by kapillondhe** [comment]
> Please add the screenshot of the test coverage and after and before changes also

@sahsisunny Added the SS of test coverage. Thanks.

**Comment by vinit717** [comment]
Please fix the tests

**Comment by kapillondhe** [comment]
> Please fix the tests

@vinit717 Just removed the extra spaces at line 386 (which were added while resolving merges), Can you please trigger the git build check once?

**Review by sahsisunny — APPROVED** [review]

File: test/unit/middlewares/tasks-validator.test.js [hunk 1]
```diff
@@ -398,4 +397,97 @@ describe("getTasks validator", function () {
...
+      expect.fail("Should not have thrown an error");
+    }
+
+    expect(nextMiddlewareSpy.callCount).to.be.equal(1);
+  });
+
+  it("should return a bad request error when empty request body is provided", async function () {
+    const invalidRequestBody = {
+      // Invalid: Missing required fields
+    };
```
Comment (vinit717) [inline test/unit/middlewares/tasks-validator.test.js:434]: ??

**Review by sahsisunny — DISMISSED** [review]

**Review by vinit717 — DISMISSED** [review]

File: test/unit/middlewares/tasks-validator.test.js [hunk 2]
```diff
@@ -398,4 +397,94 @@ describe("getTasks validator", function () {
...
+      await createTask(req, res, nextMiddlewareSpy);
+      expect.fail("Should have thrown a bad request error");
+    } catch (error) {
+      expect(error);
+    }
+
+    expect(nextMiddlewareSpy.callCount).to.be.equal(0);
+  });
+
+  it("should pass when html_url in github request body is a valid URL", async function () {
```
Comment (rohan09-raj) [inline test/unit/middlewares/tasks-validator.test.js:456]: Can we have one more test case here of failure of validation ??

**Review by vinit717 — APPROVED** [review]

**Review by rohan09-raj — APPROVED** [review]

**Review by rohan09-raj — COMMENTED** [review]
@kapillondhe Tests are failing, can you please check what's the reason of it ?

**Comment by kapillondhe** [comment]
> @kapillondhe Tests are failing, can you please check what's the reason of it ?

@rohan09-raj I checked the failing test, it is not related to the changes. it might be flaky, I ensured this by taking tests run on my machine, and all tests are passing.

**Comment by rohan09-raj** [comment]
> > @kapillondhe Tests are failing, can you please check what's the reason of it ?
> 
> @rohan09-raj I checked the failing test, it is not related to the changes. it might be flaky, I ensured this by taking tests run on my machine, and all tests are passing.

@kapillondhe Can you please trace that who has added this test and the approvers of the PRs and tag them in discord to fix this ? 
Also, you can create an issue as well and link it here so that reference is present

**Comment by kapillondhe** [comment]
> > > @kapillondhe Tests are failing, can you please check what's the reason of it ?
> > 
> > 
> > @rohan09-raj I checked the failing test, it is not related to the changes. it might be flaky, I ensured this by taking tests run on my machine, and all tests are passing.
> 
> @kapillondhe Can you please trace that who has added this test and the approvers of the PRs and tag them in discord to fix this ? Also, you can create an issue as well and link it here so that reference is present 

Created issue #1485 to track this failure.

**Review by rohan09-raj — APPROVED** [review]

**Pull request merged**